### DISCUSSION TITLE
Mappings caveat or hidded beahaviour or bug with complex json

### DISCUSSION THREAD

**Discussion by sviyh** [redpanda-data/connect#2561] [Q&A] [answered]
This works:
```
pipeline:
  processors:
    - mapping: |
        root.issue = {
          self: this.self,
          id: this.id,
          key: this.key,
        }
        root.issue.creator = this.fields.creator.displayName
```

while this doesn't:
```
pipeline:
  processors:
    - mapping: |
        root.issue = {
          self: this.self,
          id: this.id,
          key: this.key,
          creator: this.fields.creator.displayName
        }
```

It returns the error which explains nothing: "failed assignment (line 1): mapping returned invalid key type: <nil>"

**Comment by mihaitodor** [comment, marked as answer]
Hey @sviyh 👋 I'm really struggling to reproduce the issue you're seeing. At a quick glance, the bloblang code you shared above isn't valid, since the keys in the `issue` object you're trying to create aren't double quoted. For example, this works just fine:

```coffee
root.issue = {
  "self": this.self,
  "id": this.id,
  "key": this.key,
  "creator": this.fields.creator.displayName
}
```
You'd get nulls if the input message doesn't have the fields which are being referenced via `this`, but I'm not sure how to reproduce the error you're seeing.

LE: Moving to a discussion as per #2026.

**Reply by sviyh** [reply]
Hello @mihaitodor!
It worked and your exaplanation kinda works.  

The caveat still stands though - the first snippet of code still works without any quotation. Since it was working untill I've added more layers to the source json it led me to think I don't need no quatations there.

**Reply by mihaitodor** [reply]
Since you didn’t share any sample input, my guess is that you’re input message doesn’t contain a field called `creator`, which means that the unquoted `creator` resolves to `null`, which is an invalid key type. Or something like that.
